### PR TITLE
ansible: Clarify doc on owner reference injection

### PR DESCRIPTION
**Pull request by Jamstah**
**Description of the change:**
Clarify doc on owner reference injection

**Motivation for the change:**
It took me a while to understand how it worked, so I wanted to make the doc more obvious.

Ref: #1969

### CODE REVIEW

File: doc/ansible/dev/advanced_options.md [hunk 1]
```diff
@@ -41,7 +41,11 @@ Some features can be overridden per resource via an annotation on that CR. The o
 
 The ansible runner will keep information about the ansible run in the container.  This is located `/tmp/ansible-operator/runner/<group>/<version>/<kind>/<namespace>/<name>`. To learn more  about the runner directory you can read the [ansible-runner docs](https://ansible-runner.readthedocs.io/en/latest/index.html).
 
-## Turning off Dependent Watches and Owner Reference Injection
+## Owner Reference Injection
+
+Owner references enable [Kubernetes Garbage Collection](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/) to clean up after a CR is deleted. Owner references are injected by ansible operators by default by the proxy.
+
+Owner references only apply to resources in the same namespace as the CR. Resources outside the namespace of the CR will automatically be annotated with `operator-sdk/primary-resource` and `operator-sdk/primary-resource-type` to track creation. These resources will not be automatically garbage collected. To handle deletion of these resources, use a [finalizer](finalizers.md).
```
Comment (camilamacedo86) [inline doc/ansible/dev/advanced_options.md]: May it should be after the next paragraph, wdyt?

File: doc/ansible/information-flow-ansible-operator.md [hunk 2]
```diff
@@ -24,8 +24,9 @@
 
 ### The Proxy
  * Every request to the k8s api goes through the proxy.
- * The owner reference is injected into the object that is being created.
- * The proxy then adds dependent watches for the correct controller if we have not started watching the type already
+ * The owner reference is injected into the object that is being created in the same namespace as the CR.
+ * The operator-sdk annotations are injected into the object that is being created outside of namepsace of the CR.
```
Comment (camilamacedo86) [inline doc/ansible/information-flow-ansible-operator.md]: I am not sure if it makes sense for me 
What `operator-sdk annotations` are you speaking about here?

Comment (Jamstah) [reply doc/ansible/information-flow-ansible-operator.md]: The operator sdk annotations are: `operator-sdk/primary-resource` and `operator-sdk/primary-resource-type`

These are applied to k8s resources that are created in a different namespace to the namespace that the Custom Resource is created in, becuase the owner references for garbage collection wouldn't work between namespaces.

File: doc/ansible/dev/advanced_options.md (hunk 1 above)
Comment (Jamstah) [reply doc/ansible/dev/advanced_options.md]: I thought it made more sense to explain what happens, and then explain how you can change it.

**Pull request merged**